test/harnesses/StakeRegistryHarness.sol

```diff
@@ -49,35 +49,31 @@ contract StakeRegistryHarness is StakeRegistry {
             uint8(quorumNumbers[quorumNumbers.length - 1]) < quorumCount,
             "StakeRegistry._registerOperator: greatest quorumNumber must be less than quorumCount"
         );
-        OperatorStakeUpdate memory _newTotalStakeUpdate;
-        // add the `updateBlockNumber` info
-        _newTotalStakeUpdate.updateBlockNumber = uint32(block.number);
-        // for each quorum, evaluate stake and add to total stake
-        for (uint8 quorumNumbersIndex = 0; quorumNumbersIndex < quorumNumbers.length; ) {
-            // get the next quorumNumber
-            uint8 quorumNumber = uint8(quorumNumbers[quorumNumbersIndex]);
-            // evaluate the stake for the operator
-            // since we don't use the first output, this will use 1 extra sload when deregistered operator's register again
-            (, uint96 stake) = _updateOperatorStake(operator, operatorId, quorumNumber);
-            // check if minimum requirement has been met, will be 0 if not
+
+        for (uint i = 0; i < quorumNumbers.length; ) {            
+            /**
+             * Update the operator's stake for the quorum and retrieve their current stake
+             * as well as the change in stake.
+             * If this method returns `stake == 0`, the operator has not met the minimum requirement
+             * 
+             * TODO - we only use the `stake` return here. It's probably better to use a bool instead
+             *        of relying on the method returning "0" in only this one case.
+             */
```

Comment:
Question for reviewers - currently, our history gets pushed to every time we do an update, even if the last update was the same block. 

This can lead to chains in the history where multiple history updates that occur in the same block have a "nextUpdateBlockNum" field that points to the same block as the next update.

I feel like it makes more sense to just check if the last update is the current block, and update that entry's stake if so.